### PR TITLE
Add the possibility to change the orientation of the foot in the swing foot planner when the contact is not active

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
Thanks to this PR we can update the contact list in the swing foot planner when the contact is not active and the new orientation is different from the previous one.

- [x] Publish the theory implemented in the planner
- [x] update the changelog

### CODE REVIEW

File: src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp [hunk 1]
```diff
@@ -91,10 +215,12 @@ bool SO3Planner<trivialization>::evaluatePoint(const std::chrono::nanoseconds& t
...
-    // TODO (Giulio) avoid to use t / T you may use std::chrono::ratio
-    const double s = 10 * std::pow(t / T, 3)
-                     - 15 * std::pow(t / T, 4)
-                     + 6 * std::pow(t / T, 5);
+    const double s = polynomial.a0 + //
+                     polynomial.a1 * t + //
+                     polynomial.a2 * t * t + //
+                     polynomial.a3 * t * t * t + //
+                     polynomial.a4 * t * t * t * t + //
+                     polynomial.a5 * t * t * t * t * t;
```
Comment (S-Dafarra) [inline src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp:223]: I am a bit afraid of the numeric issues arising when multiplying these doubles together. 
For example, the order of evaluation is compiler dependent (https://en.cppreference.com/w/cpp/language/eval_order).

In other words, the output of ``a1*t*t`` might be different from ``a1*(t*t)``.

What about reusing the subexpressions. Something like:
```
t2 = t*t;
t3 = t2*t;
t4 =  t3*t;
t5 = t4*t;
s= a0 + a1 * t + a2*t2 + ....
```

File: src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp [hunk 2]
```diff
@@ -110,28 +236,31 @@ bool SO3Planner<trivialization>::evaluatePoint(const std::chrono::nanoseconds& t
...
+    // You can find the computation of dds in "Modern Robotics: Mechanics,
     // Planning, and Control" (Chapter 9.2)
-    const double sDdot = 10 * 3 * 2 * t / std::pow(T, 3)
-                         - 15 * 4 * 3 * std::pow(t, 2) / std::pow(T, 4)
-                         + 6 * 5 * 4 * std::pow(t, 3) / std::pow(T, 5);
-    acceleration = m_distance * sDdot;
+    const double dds = 2 * polynomial.a2 + //
+                       3 * 2 * polynomial.a3 * t + //
+                       4 * 3 * polynomial.a4 * t * t + //
+                       5 * 4 * polynomial.a5 * t * t * t;
```
Comment (S-Dafarra) [inline src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp:255]: Same as above

File: src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp (hunk 1 above)
Comment (GiulioRomualdi) [reply src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp]: Done here: https://github.com/ami-iit/bipedal-locomotion-framework/pull/688/commits/7b328423092a305b5f4c506e37a1a9bd1657cb70

File: src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp (hunk 2 above)
Comment (GiulioRomualdi) [reply src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp]: Done here: https://github.com/ami-iit/bipedal-locomotion-framework/pull/688/commits/7b328423092a305b5f4c506e37a1a9bd1657cb70

File: src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp [hunk 3]
```diff
@@ -27,12 +27,107 @@ constexpr double durationToSeconds(const std::chrono::duration<Rep, Period>& d)
...
+    double kv0, kv1, ka0, ka1;
+    if (!findVectorCoefficient(m_distance, initialCondition.velocity, kv0)
+        || !findVectorCoefficient(m_distance, initialCondition.acceleration, ka0)
+        || !findVectorCoefficient(m_distance, finalCondition.velocity, kv1)
+        || !findVectorCoefficient(m_distance, finalCondition.acceleration, ka1))
+    {
+        log()->error("{} The velocity and the boundary acceleration must be a linear dependent to "
+                     "the vector representing the axis between the two rotations.",
+                     logPrefix);
+        return false;
```
[the remark this answers is not in the document]
Comment (GiulioRomualdi) [reply src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp:108]: As discussed I implemented a function that simplifies the projection of the tangent vector https://github.com/ami-iit/bipedal-locomotion-framework/pull/688/commits/539a5288565a507728d8c37a4082c15133ead6f8

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**